packages/web/src/components/gcds-date-input/gcds-date-input.tsx

```diff
@@ -0,0 +1,542 @@
+    }
+
+    this.validateValue();
+    if (this.value && isValidDate(this.value)) {
+      this.splitFormValue();
+      this.setValue();
+
+      this.initialValue = this.value;
+    }
+  }
+
+  componentWillUpdate() {
+    if (this.validator) {
+      this._validator = getValidator(this.validator);
+    }
+  }
+
+  render() {
+    const {
+      legend,
+      name,
+      format,
+      required,
+      hint,
+      errorMessage,
+      disabled,
+      lang,
+      hasError,
+    } = this;
+
+    let requiredAttr = {};
+
+    if (required) {
+      requiredAttr['aria-required'] = 'true';
+    }
+
+    // Array of months 01 - 12
+    const options = Array.from({ length: 12 }, (_, i) =>
+      i + 1 < 10 ? `0${i + 1}` : `${i + 1}`,
+    );
+
+    const month = (
+      <gcds-select
+        label={i18n[lang].month}
+        selectId="month"
+        name="month"
+        defaultValue={i18n[lang].selectmonth}
+        disabled={disabled}
+        onInput={e => this.handleInput(e, 'month')}
+        onChange={e => this.handleInput(e, 'month')}
+        value={this.monthValue}
+        class={`gcds-date-input__month ${hasError['month'] ? 'gcds-date-input--error' : ''}`}
+        {...requiredAttr}
+        aria-invalid={hasError['month'].toString()}
+        aria-description={hasError['month'] && errorMessage}
+      >
+        {options.map(option => (
+          <option key={option} value={option}>
+            {i18n[lang]['months'][option]}
+          </option>
+        ))}
+      </gcds-select>
+    );
+
+    const year = (
+      <gcds-input
+        name="year"
+        label={i18n[lang].year}
+        inputId="year"
+        type="number"
+        size={4}
+        disabled={disabled}
+        value={this.yearValue}
+        autocomplete="year"
+        onInput={e => this.handleInput(e, 'year')}
+        onChange={e => this.handleInput(e, 'year')}
+        class={`gcds-date-input__year ${hasError['year'] ? 'gcds-date-input--error' : ''}`}
+        {...requiredAttr}
+        aria-invalid={hasError['year'].toString()}
+        aria-description={hasError['year'] && errorMessage}
+      ></gcds-input>
+    );
+
+    const day = (
+      <gcds-input
+        name="day"
+        label={i18n[lang].day}
+        inputId="day"
+        type="number"
+        size={2}
+        disabled={disabled}
+        value={this.dayValue}
+        autocomplete="day"
+        onInput={e => this.handleInput(e, 'day')}
+        onChange={e => {
+          this.handleInput(e, 'day');
+          this.formatDay(e);
+        }}
+        class={`gcds-date-input__day ${hasError['day'] ? 'gcds-date-input--error' : ''}`}
+        {...requiredAttr}
+        aria-invalid={hasError['day'].toString()}
+        aria-description={hasError['day'] && errorMessage}
+      ></gcds-input>
+    );
+
+    return (
+      <Host name={name} onBlur={() => this.onBlur()}>
+        {this.validateRequiredProps() && (
+          <gcds-fieldset
+            legend={legend}
```

Comment:
I know we had previously agreed to call this prop `legend` because it is a legend, but after all the discussions we had about fieldsets in the last week, I'm wondering if it would make more sense to name it `label` to keep it consistent (less confusing?) with other form fields for our users. What do you think @daine @ethanWallace ?

Reply:
I'm now wondering if it's best to wait on the final decision for the change to the fieldset component for this. But yes I'm leaning towards the fact that it's probably appropriate to call it a `label` because that's what it is in this case. From what the designers are also proposing, this label also looks different and will have the `medium` style as opposed to `bold`. I will tag you in the slack thread where they mentioned it

Reply:
Just flagging (I don't disagree): If we wait for the final decision on the fieldset component, we will need to hold releasing this component until then. Since the guidance still seems to be requiring some time, it wouldn't be the worst but Abdul did say that one team was waiting on the release of this component.